t3f/shapes.py

```diff
@@ -133,6 +148,25 @@ def lazy_raw_shape(tt):
     return raw_shape(tt)
 
 
+def lazy_batch_size(tt):
```

Comment:
Because it returns the data on the compilation stage if available, thus avoiding doing any real work. Do you have better ideas for the name?)